wger/nutrition/views/meal_item.py

```diff
@@ -78,16 +84,31 @@ def get_context_data(self, **kwargs):
         Send some additional data to the template
         '''
         context = super(MealItemCreateView, self).get_context_data(**kwargs)
-        context['form_action'] = reverse('nutrition:meal_item:add',
-                                         kwargs={'meal_id': self.meal.id})
-        context['ingredient_searchfield'
-                ] = self.request.POST.get('ingredient_searchfield', '')
+
+        if self.meal_id:
+            context['form_action'] = reverse('nutrition:meal_item:add',
+                                             kwargs={'meal_id': self.meal.id})
+        else:
+            context['form_action'] = reverse('nutrition:meal_item:add-meal',
+                                             kwargs={'plan_pk': self.plan_pk})
+        context['ingredient_searchfield'] =\
+            self.request.POST.get('ingredient_searchfield', '')
+        context['plan_pk'] = self.plan_pk
         return context
 
     def form_valid(self, form):
         '''
         Manually set the corresponding meal
         '''
+        if not self.meal_id:
+            plan = get_object_or_404(
+                NutritionPlan, pk=self.plan_pk, user=self.request.user)
+            meal = Meal.objects.create(plan=plan, order=1)
+            self.meal = meal
+            time = form.cleaned_data['time']
+            self.meal.time = time
+            self.meal.save()
```

Comment:
Kindly assign the variable directly. This will make the code more concise.

Reply:
Thanks for the feedback. Done ✅